### DISCUSSION TITLE
Offer "gettext style" procedures for static copy.

### DISCUSSION THREAD

**Discussion by angrybrad** [craftcms/cms#9460] [Ideas]
> Created by: Howard Roark (howie@codingcoop.org) on 2015/08/11 14:11:50 +0000
> Votes at time of UserVoice import: 2

-----------------------------------

In the past I have used what I would describe as "gettext style" procedures.   The basic idea is that on the "first go" you fill your template with the intended copy and translation tags (which is already the case with Craft). After you finish up a parsing tool combs through to find all the places where they exist and generates a ".pot" file.  This file is basically a translation template for your copy which works with a variety of GUI editors.  A translator take the file, fill in all the copy of a desired locale and pass it back as a key/value store ".po" file.

The reason this is great is because it allows you to place all of your copy over time and then get it all back when you wrap up.  Currently you have to build your own static translation file with Craft.

As I see it there are two options...

1. Build in a proper standardized gettext setup for static translations.

2.  Instead of keeping static translations in a file, store them in the db.  Then offer a cms page specifically specifically for entering translations using the same methodology.

**Comment by angrybrad** [comment]
> Posted by Wesley B (wesleyb@phmgmt.com) on 2016/09/01 16:40:23 +0000
-----------------------------------

There is also another post about supporting PO files: #970

I would also like to see support for `gettext` / `.po`; this would allow automation of translations using a standard. Here are three companies that work with `.po` files from digital sources (like app and websites):

- CrowdIn: https://crowdin.com/
- LingoTek: https://lingotek.com/
- SpeakLike: http://www.speaklike.com/

**Comment by angrybrad** [comment]
> Posted by Christian (seelbach@gmx.de) on 2015/03/09 00:49:13 +0000
-----------------------------------

You can already do this with this plugin: https://github.com/boboldehampsink/translate

**Comment by kruncher** [comment]
The above plugin is not compatible with Craft 3.

**Comment by brandonkelly** [comment]
Enupal Translate might be a suitable substitute? https://plugins.craftcms.com/enupal-translate
